src/ucp/tag/tag_send.c

```diff
@@ -179,14 +179,15 @@ ucp_tag_send_req(ucp_request_t *req, size_t count, ssize_t max_short,
         return UCS_STATUS_PTR(UCS_ERR_INVALID_PARAM);
     }
 
-    ucp_send_req_stat(req);
-
     /*
      * Start the request.
      * If it is completed immediately, release the request and return the status.
      * Otherwise, return the request.
      */
     status = ucp_request_start_send(req);
+
+    ucp_send_req_stat(req);
```

Comment:
it's needed to keep statistics workable because RNDV initialization is delayed a bit.